packages/core/types/src/types/core/config/index.ts

```diff
@@ -0,0 +1,5 @@
+export type { Server } from './server';
```

Comment:
This currently seems to cause an issue with lint thinking there are duplicate Server namespaces, and the workaround in middlewares doesn't seem to work.

We may want to go with ServerInput, AdminInput, ApiInput, etc which is a pattern we've also used with RouteInput.